litebird_sim/scanning.py

```diff
@@ -0,0 +1,399 @@
+# -*- encoding: utf-8 -*-
+
+from astropy.coordinates import ICRS, get_body_barycentric, BarycentricMeanEcliptic
+from astropy.time import Time
+import astropy.units as u
+from numba import njit
+import numpy as np
+
+
+YEARLY_OMEGA_SPIN_HZ = 2 * np.pi / (1.0 * u.year).to(u.s).value
+
+
+@njit
+def qrotation_x(theta):
+    return (np.sin(theta / 2), 0.0, 0.0, np.cos(theta / 2))
+
+
+@njit
+def qrotation_y(theta):
+    return (0.0, np.sin(theta / 2), 0.0, np.cos(theta / 2))
+
+
+@njit
+def qrotation_z(theta):
+    return (0.0, 0.0, np.sin(theta / 2), np.cos(theta / 2))
+
+
+@njit
+def quat_right_multiply(result, other_v1, other_v2, other_v3, other_w):
```

Comment:
Allow `result` and `other` (I use the entire a quaternion instead of the individual components) to be arbitrary stacks of quaternions the following code should do the job
```python
    v1 = (
        result[..., 3] * other[..., 0]
        + result[..., 0] * other[..., 3]
        + result[..., 1] * other[..., 2]
        - result[..., 2] * other[..., 1]
    )

.......

    result[..., 0] = v1
    result[..., 1] = v2
    result[..., 2] = v3
    result[..., 3] = w
```

Reply:
In this example `result` can have any shape `(a, b, 4)` and `other` can have any shape broadcastable to the one of `result`